app/src/main/java/com/sunasterisk/boringweather/ui/search/SearchFragment.kt

```diff
@@ -0,0 +1,75 @@
+package com.sunasterisk.boringweather.ui.search
+
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.setFragmentResult
+import com.sunasterisk.boringweather.R
+import com.sunasterisk.boringweather.base.BaseFragment
+import com.sunasterisk.boringweather.data.model.City
+import com.sunasterisk.boringweather.di.Injector
+import com.sunasterisk.boringweather.ui.main.findNavigator
+import com.sunasterisk.boringweather.util.showToast
+import kotlinx.android.synthetic.main.fragment_search.*
+
+class SearchFragment : BaseFragment(), SearchContract.View {
+    override val layoutResource: Int = R.layout.fragment_search
+
+    override var presenter: SearchContract.Presenter? = null
+    override fun showSearchResult(cities: List<City>) {
+        (recyclerSearchCity.adapter as? CityAdapter)?.submitList(cities)
+    }
+
+    override fun showError(errorStringRes: Int) {
+        requireContext().showToast(getString(errorStringRes))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        presenter = SearchPresenter(this, Injector.getCityRepository(requireContext()))
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initView()
+        requestEditTextSearchFocus()
+    }
+
+    private fun initView() {
+        textField.setStartIconOnClickListener {
```

Comment:
xử lý sự kiện nên tách riêng khỏi initView ra hàm khác, vì sau này hàm sẽ bị phình to đấy. 
(Recommend 7-10 dòng cho 1 function)